pkgs/test/buildFHSEnv/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, buildFHSEnv
+, runCommand
+, glibc
+, callPackage
+}:
+
+let
+  makeSharedObjectTest = sharedObject: targetPkgs: let
+    lddFHSEnv = buildFHSEnv {
+      name = "ldd-with-ncurses-FHS-env";
+      inherit targetPkgs;
+      runScript = "ldd";
+    };
+    ldd-in-FHS = "${lddFHSEnv}/bin/${lddFHSEnv.name}";
```

Comment:
This is fine but we should probably just provide buildFHSEnv outputs a meta.mainProgram that internalises this logic.

Reply:
> This is fine but we should probably just provide buildFHSEnv outputs a meta.mainProgram that internalises this logic.

I'm not sure I understand - do you want to use:

```nix
"${lddFHSEnv}/bin/${lddFHSEnv.meta.mainProgram}"
```

?

Reply:
Ah, indeed `mainProgram` is only the exe name; I got confused.

Though thinking about it again, with  `mainProgram` we could use `getExe` here. That's a discussion for another PR though.